clients/fides-js/src/fides.ts

```diff
@@ -78,8 +78,18 @@ const updateExperience: UpdateExperienceFn = ({
 /**
  * Initialize the global Fides object with the given configuration values
  */
-async function init(this: FidesGlobal, config: FidesConfig) {
-  this.config = config;
+async function init(this: FidesGlobal, config?: FidesConfig) {
+  // confused by the "this"? see https://www.typescriptlang.org/docs/handbook/2/functions.html#declaring-this-in-a-function
+
+  if (!config && !this.config) {
```

Comment:
You're checking for the same conditions twice here which suggests that this could be simplified further. I think this is equivalent:

```ts
async function init(this: FidesGlobal, providedConfig?: FidesConfig) {
  const config = (this.config ??= providedConfig) ?? raise("Fides must be initialized with a configuration object");
  // ...
}
```

Where `raise` is a tiny helper to allow throwing errors in an expression context. `throw` is a statement so we can't use it like`foo ?? throw "Trouble!"`

Reply:
Thanks! Updated to implement. It's high time we added a common util file also!